rotkehlchen/tests/unit/test_tokens.py

```diff
@@ -82,33 +86,57 @@ def test_detect_tokens_for_addresses(rotkehlchen_api_server, ethereum_accounts):
     tokens = rotki.chains_aggregator.ethereum.tokens
     tokens.evm_inquirer.multicall = MagicMock(side_effect=tokens.evm_inquirer.multicall)
     Inquirer.find_usd_prices = MagicMock(side_effect=Inquirer.find_usd_prices)
-    with patch(
-        'rotkehlchen.globaldb.handler.GlobalDBHandler.get_token_detection_data',
-        side_effect=lambda *args, **kwargs: ([  # mock the returned list to avoid changing this test with every assets version  # noqa: E501
-            EvmTokenDetectionData(
-                identifier=A_WETH.identifier,
-                address=string_to_evm_address('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'),
-                decimals=18,
-            ), EvmTokenDetectionData(
-                identifier=A_LPT.identifier,
-                address=string_to_evm_address('0x58b6A8A3302369DAEc383334672404Ee733aB239'),
-                decimals=18,
-            ), EvmTokenDetectionData(
-                identifier=A_OMG.identifier,
-                address=string_to_evm_address('0xd26114cd6EE289AccF82350c8d8487fedB8A0C07'),
-                decimals=18,
-            ), EvmTokenDetectionData(
-                identifier=A_DAI.identifier,
-                address=string_to_evm_address('0x6B175474E89094C44Da98b954EedeAC495271d0F'),
-                decimals=18,
-            ),
-        ], []),
+    with (
+        patch('rotkehlchen.chain.ethereum.tokens.EthereumTokens._detect_tokens', side_effect=mock_eth_detect_tokens) as detect_tokens_patch,  # noqa: E501
+        patch(
+            'rotkehlchen.globaldb.handler.GlobalDBHandler.get_token_detection_data',
+            side_effect=lambda *args, **kwargs: ([  # mock the returned list to avoid changing this test with every assets version  # noqa: E501
+                EvmTokenDetectionData(
+                    identifier=A_WETH.identifier,
+                    address=string_to_evm_address('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'),
+                    decimals=18,
+                ), EvmTokenDetectionData(
+                    identifier=A_LPT.identifier,
+                    address=string_to_evm_address('0x58b6A8A3302369DAEc383334672404Ee733aB239'),
+                    decimals=18,
+                ), EvmTokenDetectionData(
+                    identifier=A_OMG.identifier,
+                    address=string_to_evm_address('0xd26114cd6EE289AccF82350c8d8487fedB8A0C07'),
+                    decimals=18,
+                ), EvmTokenDetectionData(
+                    identifier=A_DAI.identifier,
+                    address=string_to_evm_address('0x6B175474E89094C44Da98b954EedeAC495271d0F'),
+                    decimals=18,
+                ),
+            ], []),
+        ),
     ):
         detection_result = tokens.detect_tokens(False, [addr1, addr2, addr3])
+
+    assert detect_tokens_patch.call_count == 2  # once for normal addresses, once for the proxy
+    assert detect_tokens_patch.call_args_list[1].kwargs['addresses'] == [addr3_proxy]
```

Comment:
could you add in a small comment why this checks? I'm sure that if I had to adjust the test in the future I would get a bit lost just with this